wasm/src/main/java/com/dylibso/chicory/wasm/io/InputStreamWasmInputStream.java

```diff
@@ -0,0 +1,142 @@
+package com.dylibso.chicory.wasm.io;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.VarHandle;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+/**
+ * A WASM input stream based on {@link InputStream}.
+ */
+class InputStreamWasmInputStream extends WasmInputStream {
+    private static final VarHandle le32 =
```

Comment:
What would be the alternative to the usage of `VarHandle` / `MethodHandles` ?

Those classes are not going to be available on some JVM targets(i.e. Android) and I refrain from using them in such a low-level building block.

If I understand correctly an alternative would be to use `ByteBuffer` primitives, such as:

```java
ByteBuffer.wrap(byteArray)
     .order(ByteOrder.BIG_ENDIAN)
     .asIntBuffer(); // .asLongBuffer()
```

Unless you have a really compelling argument for moving ahead with this implementation I kindly ask you to avoid any usage of `java.lang.invoke`, especially in the low-level `wasm` sub-module.

Reply:
According to what I could find online, Android does in fact support `java.lang.invoke`, at least to some extent (I was not able to determine exactly what).

In any event, when it comes to Android I think we need to do one of two things starting immediately:

* Start testing with Android to ensure that it really is supported, or else
* Stop saying we will support Android

Reading online I found that Android does in fact support `j.l.i`... and, they also support Java 17 starting with Android 14. So it seems like we may be doing some acrobatics for no good reason.

Now, that having been said: the reason for this specific implementation choice is that with only one static instance, there is no allocation pressure; this is the recommended approach for such things (it's the reason this API exists). Allocating a byte buffer each time increases allocation pressure and incurs some overhead due to the extra setup every time we want to read a value, or at *best*, every time we set up the stream. The alternative would be to do the math inline to combine the bytes, and hope that HotSpot will eventually recognize the pattern and optimize it for us, but I know of no evidence that it would do so.

Reply:
Thanks, appreciated 👍